custom_components/tuya_local/devices/beca_bht002_thermostat_c.yaml

```diff
@@ -53,9 +53,17 @@ primary_entity:
       name: floor_temperature
       mapping:
         - scale: 2
+    - id: 103
+      type: enum
+      name: tempSwitch

```

Comment:
Thank you for looking into this, and can you help me edit my code commit? It’s my first commit, and I can’t figure out how to change it again. :)

I agree the name should be changed to `hvac_action`, and the mapping to `heating` and `idle` makes more sense.

Regarding marking it optional, I’m not entirely sure what effect that has. However, considering that DP unknown_104 likely isn’t available on the same type of thermostats without an external thermocouple, making it optional could make sense if it allows enabling or disabling it during setup based on the specific hardware configuration.

Reply:
Marking it optional just lets devices match this config if it is missing in the data they send.  In that case, hvac_action will report as None (null), which is the same as if it is not defined in the config, and HA does not show anything.

This was issue #59 which was an early device addition, and we hadn't yet figured out the best guidance for getting dps info, so the dps info was just taken from a forum post which did not mention dp 103. And later requests to add the heating state ended up closed because the users could not identify any dp that was giving the info, so I think those older devices does not report dp 103. On the same forum a user apparently found instructions that a resistor needs to be added to the PCB for the heating state to even be available to the firmware, so it must even be a hardware revision.

I think dp 104 is always available, it is dp 102 that becomes unavailable when it is indicating False.  But knowing this, we can probably now merge moes_bht002_thermostat_c.yaml, and beca_bht002_thermostat_c.yaml, which differ only in whether that external temperature probe is there or not.

Reply:
Pushing additional changes to the same branch should update the PR. But I can edit this one, as I have a clear picture of the changes, and perhaps you are still not entirely clear on them all.

Reply:
Thanks for support!